packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,672 @@
+        assert.isValidBaseUnitAmount('takerAssetFillAmount', takerAssetFillAmount);
+        const data = this._transactionEncoder.fillOrderNoThrowTx(signedOrder, takerAssetFillAmount);
+        return this._handleFillAsync(data, takerAddress, signedOrder.feeRecipientAddress, orderTransactionOpts);
+    }
+
+    /**
+     * Attempts to fill a specific amount of an order. If the entire amount specified cannot be filled,
+     * the fill order is abandoned.
+     * @param   signedOrder          An object that conforms to the SignedOrder interface.
+     * @param   takerAssetFillAmount The amount of the order (in taker asset baseUnits) that you wish to fill.
+     * @param   takerAddress         The user Ethereum address who would like to fill this order. Must be available via the supplied
+     *                               Provider provided at instantiation.
+     * @param   orderTransactionOpts Optional arguments this method accepts.
+     * @return  Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async fillOrKillOrderAsync(
+        signedOrder: SignedOrder,
+        takerAssetFillAmount: BigNumber,
+        takerAddress: string,
+        orderTransactionOpts: OrderTransactionOpts = { shouldValidate: true },
+    ): Promise<string> {
+        assert.doesConformToSchema('signedOrder', signedOrder, schemas.signedOrderSchema);
+        assert.isValidBaseUnitAmount('takerAssetFillAmount', takerAssetFillAmount);
+        const data = this._transactionEncoder.fillOrKillOrderTx(signedOrder, takerAssetFillAmount);
+        return this._handleFillAsync(data, takerAddress, signedOrder.feeRecipientAddress, orderTransactionOpts);
+    }
+
+    /**
+     * Batch version of fillOrderAsync. Executes multiple fills atomically in a single transaction.
+     * @param   signedOrders          An array of signed orders to fill.
+     * @param   takerAssetFillAmounts The amounts of the orders (in taker asset baseUnits) that you wish to fill.
+     * @param   takerAddress          The user Ethereum address who would like to fill these orders. Must be available via the supplied
+     *                                Provider provided at instantiation.
+     * @param   orderTransactionOpts  Optional arguments this method accepts.
+     * @return  Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async batchFillOrdersAsync(
+        signedOrders: SignedOrder[],
+        takerAssetFillAmounts: BigNumber[],
+        takerAddress: string,
+        orderTransactionOpts: OrderTransactionOpts = { shouldValidate: true },
+    ): Promise<string> {
+        assert.doesConformToSchema('signedOrders', signedOrders, schemas.signedOrdersSchema);
+        for (const takerAssetFillAmount of takerAssetFillAmounts) {
+            assert.isBigNumber('takerAssetFillAmount', takerAssetFillAmount);
+        }
+        const feeRecipientAddress = getFeeRecipientOrThrow(signedOrders);
```

Comment:
There can be more then one `feeRecipientAddress` in the supplied orders. If there are multiple coordinator server involved in a batch fill, the client must send a request to every coordinator server and must retrieve a signature from each. Sorry if this was not clear.